statistics/handle/statscache.go

```diff
@@ -39,47 +72,33 @@ func (key statsCacheKey) Hash() []byte {
 	return buf
 }
 
-// newStatsCache returns a new statsCache with capacity maxMemoryLimit.
-func newStatsCache(memoryLimit int64) *statsCache {
-	// Since newStatsCache controls the memory usage by itself, set the capacity of
+func newSimpleLRUStatsCache(memoryCapacity int64) *simpleLRUStatsCache {
+	// since stats cache controls the memory usage by itself, set the capacity of
 	// the underlying LRUCache to max to close its memory control
-	cache := kvcache.NewSimpleLRUCache(uint(memoryLimit), 0.1, 0)
-	c := statsCache{
+	cache := kvcache.NewSimpleLRUCache(uint(memoryCapacity), 0.1, 0)
+	c := simpleLRUStatsCache{
 		cache:       cache,
-		memCapacity: memoryLimit,
-		memTracker:  memory.NewTracker(memory.LabelForStatsCache, -1),
+		memCapacity: memoryCapacity,
+		memTracker:  memory.NewTracker(memory.LabelForStatsCache, memoryCapacity),
 	}
 	return &c
 }
 
-// Clear clears the statsCache.
-func (sc *statsCache) Clear() {
-	// Since newStatsCache controls the memory usage by itself, set the capacity of
-	// the underlying LRUCache to max to close its memory control
+// SetBytesLimit sets the bytes limit for this tracker.
+func (sc *simpleLRUStatsCache) SetBytesLimit(BytesLimit int64) {
 	sc.mu.Lock()
 	defer sc.mu.Unlock()
-	cache := kvcache.NewSimpleLRUCache(uint(sc.memCapacity), 0.1, 0)
-	sc.memTracker.ReplaceBytesUsed(0)
-	sc.cache = cache
-	sc.version = 0
+	sc.memTracker.SetBytesLimit(BytesLimit)
+	sc.memCapacity = BytesLimit
 }
 
-// GetAll get all the tables point.
-func (sc *statsCache) GetAll() []*statistics.Table {
-	sc.mu.Lock()
-	defer sc.mu.Unlock()
-	values := sc.cache.GetAll()
-	tables := make([]*statistics.Table, 0)
-	for _, v := range values {
-		if t, ok := v.(*statistics.Table); ok && t != nil {
-			tables = append(tables, t)
-		}
-	}
-	return tables
+// BytesConsumed returns the consumed memory usage value in bytes.
+func (sc *simpleLRUStatsCache) BytesConsumed() int64 {
```

Comment:
Lock()?

Reply:
No need for this function, it's atomic value.